Library/Homebrew/extend/pathname.rb

```diff
@@ -423,15 +423,19 @@ def env_script_all_files(dst, env)
       script_name:  T.any(String, Pathname),
       java_opts:    String,
       java_version: T.nilable(String),
+      env:          T::Hash[String, String],
     ).returns(Integer)
   }
-  def write_jar_script(target_jar, script_name, java_opts = "", java_version: nil)
+  def write_jar_script(target_jar, script_name, java_opts = "", java_version: nil, env: {})
+    env_export = +""
+    env.each { |key, value| env_export << "#{key}=\"#{value}\" " }
+
     mkpath
-    (self/script_name).write <<~EOS
+    (self/script_name).write <<~SH
       #!/bin/bash
       export JAVA_HOME="#{Language::Java.overridable_java_home_env(java_version)[:JAVA_HOME]}"
-      exec "${JAVA_HOME}/bin/java" #{java_opts} -jar "#{target_jar}" "$@"
-    EOS
+      #{env_export}exec "${JAVA_HOME}/bin/java" #{java_opts} -jar "#{target_jar}" "$@"
```

Comment:
I think either:
```bash
env ENV=var exec
```
or
```
export ENV=var
exec
```
are needed here for valid syntax.

Either way, please ensure you've tested this and found it to be working locally before marking this as non-draft